polaris-react/src/components/IndexTable/IndexTable.tsx

```diff
@@ -50,6 +59,21 @@ export interface IndexTableBaseProps {
   paginatedSelectAllActionText?: string;
   lastColumnSticky?: boolean;
   selectable?: boolean;
+  /** List of booleans, which maps to whether sorting is enabled or not for each column. Defaults to false for all columns.  */
+  sortable?: boolean[];
+  /**
+   * The direction to sort the table rows on first click or keypress of a sortable column heading. Defaults to ascending.
+   * @default 'ascending'
+   */
+  defaultSortDirection?: IndexTableSortDirection;
+  /** The current sorting direction. */
+  sortDirection?: IndexTableSortDirection;
+  /**
+   * The index of the heading that the table rows are sorted by.
+   */
+  sortColumnIndex?: number;
```

Comment:
Curious if this should be handled internally rather than set by the props? Looks like the `DataTable` does an `initialSortColumnIndex` then handles subsequent column index changes internally. Thoughts on this pattern?

Reply:
@samrose3 We need to expose this in this way because the logic that does all the sorting of the data has to happen outside of the IndexTable itself, as you can also sort an IndexTable from an external component. You can see here on the [Figma](https://www.figma.com/file/bZb9bEHNCEMiYjzO0fhK4J/Index-filters-and-search?node-id=670%3A176130) that we have a Sort button that lives within the new filtering UI that also controls the sorting state. Internalizing the sorting state within the IndexTable would make it impossible, or certainly impractical, to manage this state.